### DISCUSSION TITLE
How to get process environment variable in Vue Interface ?

### DISCUSSION THREAD

**Discussion by sandyksit** [directus/directus#10724] [Archive]
Hi Team, @mcarlson 
I have declare variable in .env file, but i am not able to get process env variables in Vue interface. Could you please help me.

**Comment by rijkvanzanten** [comment]
I'm assuming you're talking about the Directus admin app, in which case this is correct. The environment variables from the server aren't available in the admin app for security reasons (you wouldn't want to leak your database credentials etc to the public)

**Reply by sandyksit** [reply]
@rijkvanzanten, Is there any way how to set/get variables as per env like (UAT/TEST) ?

**Reply by rijkvanzanten** [reply]
On the app side?

**Reply by sandyksit** [reply]
@rijkvanzanten yes I need variable into extensions/interface.

**Reply by rijkvanzanten** [reply]
There's no built-in way to get environment variables from the API to the app. You could create a custom endpoint that returns that information, but make sure you don't accidentally leak any private data from the server 👍🏻

**Reply by sandyksit** [reply]
@rijkvanzanten I have created custom endpoint but its not calling is there any configuration

./extentions/endpoints/src/index.js

`export default (router, {env}) => {
	router.get('/', (req, res) => {
        console.log("*********************env*********************", env)
        res.send(env)
    });
};
`

and trying to access in interface.. like this
`this.api.get(`/`).then((res) => {
			console.log("res", res)
		});`

But not able to get the response

**Reply by azrikahar** [reply]
based on your example code, since your custom endpoint is in the `src` folder (`src` as in `./extensions/endpoints/src`), you should fetch it at `/src` instead of `/` 👍

**Reply by sandyksit** [reply]
@azrikahar @rijkvanzanten 

```js
export default (router, {env}) => { router.get('/extTest', (req, res) => { console.log("*********************env*********************", env) res.send('test data') }); };
```

and trying to access in interface.. like this

```js
thi.api.get('/src/extTest').then((res) => { console.log("res", res) });
```

I have used like this as per your suggestions, But every time getting html page in response.

**Reply by azrikahar** [reply]
You can try using `res.json()` instead of `res.send()` to ensure it's a JSON response. You can learn more about the difference by going through [res.send()](https://expressjs.com/en/api.html#res.send) and [res.json()](https://expressjs.com/en/api.html#res.json) in Express' official docs.

**Reply by sandyksit** [reply]
@azrikahar @rijkvanzanten, i am getting 404, http://localhost:8055/src/extTest 404 (Not Found)

**Reply by azrikahar** [reply]
@sandyksit I'd recommend sharing what you currently have in our [community Discord server](https://directus.chat) so that it may be easier for others to help verify what could be the issue with your current setup here. 👍 That said, it may also help to do some sanity checks such as:

- ensuring you are inserting the built `index.js` in the dist folder of your custom endpoint project into the `./extensions/endpoints/src/index.js` directory of the Directus project. Your custom endpoint project _should not_ be inside your Directus project.
- test the custom endpoint on your browser rather than relying on the custom interface to access it, just to ensure that endpoint exists.

**Reply by sandyksit** [reply]
Thanks, now i am able to access.
